riscv/processor.cc

```diff
@@ -189,11 +189,13 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   prv = PRV_M;
   v = false;
   csrmap[CSR_MISA] = misa = std::make_shared<misa_csr_t>(proc, CSR_MISA, max_isa);
-  csrmap[CSR_MSTATUS] = mstatus = std::make_shared<mstatus_csr_t>(proc, CSR_MSTATUS);
+  mstatus = std::make_shared<mstatus_csr_t>(proc, CSR_MSTATUS);
 
   if (xlen == 32) {
-    const reg_t mstatush_mask = MSTATUSH_MPV | MSTATUSH_GVA | MSTATUSH_SBE | MSTATUSH_MBE;
-    csrmap[CSR_MSTATUSH] = std::make_shared<rv32_high_csr_t>(proc, CSR_MSTATUSH, mstatush_mask, mstatus);
+    csrmap[CSR_MSTATUS] = std::make_shared<rv32_low_csr_t>(proc, CSR_MSTATUS, mstatus);
+    csrmap[CSR_MSTATUSH] = mstatush = std::make_shared<rv32_high_csr_t>(proc, CSR_MSTATUSH, mstatus);
+  } else {
```

Comment:
The mask for mstatush for RV32 is a little different from the high half of mstatus in RV64. Maybe we need a mask here to mask the unused fields in mstatush before writing to the whole mstatus.

Reply:
AFAICT the only differences are:
* SD, which is never writable
* SXL and UXL, which are not writable today in Spike, and even if Spike was enhanced to support that, those fields would/could/should be masked to 0 when MXLEN=32.

I have tested writing 0xffffffff to `mstatush` and I can see that it writes only the two hypervisor bits (MPV and GVA), as expected.

So I don't think any mask is needed.

Reply:
Yeah. A mistake myself. I didn't realize SXL and UXL are not writable currently.